compiler/rustc_type_ir/src/interner.rs

```diff
@@ -28,28 +29,31 @@ pub trait Interner:
     + IrPrint<FnSig<Self>>
 {
     type DefId: Copy + Debug + Hash + Eq;
-    type DefiningOpaqueTypes: Copy + Debug + Hash + Default + Eq + TypeVisitable<Self>;
     type AdtDef: Copy + Debug + Hash + Eq;
 
-    type GenericArgs: GenericArgs<Self>;
+    type GenericArgs: GenericArgs<Self> + Default;
```

Comment:
should `Default` be a super trait of `GenericArgs`?

Reply:
Yeah, I must have missed that.